### PR TITLE
Select: fix an illegal filter regexp bug

### PR DESCRIPTION
**Pull request by Leopoldthecoder**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's [Contributing Guide](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.md).
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by Leopoldthecoder** [comment]
#770

File: packages/select/src/option.vue [hunk 1]
```diff
@@ -97,7 +97,8 @@
       },
 
       queryChange(query) {
-        this.queryPassed = new RegExp(query, 'i').test(this.currentLabel);
+        let parsedQuery = query.replace(/(\^|\(|\)|\[|\]|\$|\*|\+|\.|\?|\\|\{|\}|\|)/g, '\\$1');
+        this.queryPassed = new RegExp(parsedQuery, 'i').test(this.currentLabel);
```
Comment (baiyaaaaa) [inline packages/select/src/option.vue]: queryPassed 做控制显示属性拓展性不太好，建议用 'visible'

File: packages/select/src/option.vue [hunk 2]
```diff
@@ -97,7 +97,8 @@
       },
 
       queryChange(query) {
-        this.queryPassed = new RegExp(query, 'i').test(this.currentLabel);
+        let parsedQuery = query.replace(/(\^|\(|\)|\[|\]|\$|\*|\+|\.|\?|\\|\{|\}|\|)/g, '\\$1');
```
Comment (furybean) [inline packages/select/src/option.vue]: 怎么感觉这个正则很难维护…

File: packages/select/src/option.vue (hunk 1 above)
Comment (Leopoldthecoder) [inline packages/select/src/option.vue]: 改了

**Pull request merged**